order_test.go

```diff
@@ -0,0 +1,40 @@
+package apiutil
+
+import (
+	"net/http/httptest"
+	"testing"
+)
+
+func TestOrderByFromRequest(t *testing.T) {
+	cases := []struct {
+		description string
+		query       string
+		expNumber   int
+		expOrderBy  string
+	}{
+		{"no orderBy specified", "", 0, ""},
+		{"orderBy with no direction (defaults to desc)", "title", 1, "title,desc"},
+		{"invalid orderBy", "title,asc,date", 0, ""},
+		{"orderBy with multiple parameters", "title,asc,date,desc", 2, "title,asc,date,desc"},
```

Comment:
what happens if I do `"title, date"`? Do I get `title,asc,date,asc`? Illegal?

Reply:
It makes a key " date" which you can but don't have to process wherever this is used. I could go and start stripping weird signs but that's and endless thing. Maybe just whitespace front/back?